### PR TITLE
Change races to activities

### PR DESCRIPTION
**Pull request by Mathieu414**
See the project item for the idea of the changes.

Quite a lot of things. Changed a bit how the data is displayed in the activities accordion, so that it becomes clearer what is the event description and what is your own registration.

### CODE REVIEW

**Comment by Atmos4** [comment]
@Mathieu414 you might want to rebase to master with this branch! You can also either worktree or stash our progress to make it easier to review, and create another work item!

**Comment by Atmos4** [comment]
I don't like the current data migration:
- it has var-dumps. Please if you want to display things, format them in a readable way
- it does not work and throws errors. Clean it up before the merge
- it is non revertible. Please implement a "revert" button, or better yet use the built in migrations from doctrine!

_If you want to generate an empty migration file with Doctrine, you can use the following command:_
```bash
php bin/doctrine migrations:generate
```

File: database/race_to_activity.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        $newEntry = new ActivityEntry();
+        $newEntry->user = em()->getReference(User::class, $entry[0]['user_id']);
+        # make the link between the race and the corresponding activity
+        $race = em()->createQuery("SELECT r FROM Race r WHERE r.id = :raceId")->setParameter('raceId', $entry[0]['race_id'])->getArrayResult()[0];
+        # match by name and date
+        $activity_id = em()->createQuery("SELECT a.id FROM Activity a WHERE a.name = :name AND a.date = :date")
+            ->setParameters([
+                'name' => $race["name"],
+                'date' => $race["date"]
+            ])->getSingleScalarResult();
```
Comment (Atmos4) [inline database/race_to_activity.php:41]: Wow I did not see that. 2 queries per entry is not going to work here.

It sounds like you need help with this. I can take care of it

File: database/models/categories.db.php [hunk 2]
```diff
@@ -27,8 +24,8 @@ class Category
     #[Column]
     public bool $removed = false;
 
-    /** @var Collection<int,RaceEntry> entries */
-    #[OneToMany(targetEntity: RaceEntry::class, mappedBy: "category", cascade: ["remove"])]
+    /** @var Collection<int,ActivityEntry> entries */
+    #[OneToMany(targetEntity: ActivityEntry::class, mappedBy: "category", cascade: ["remove"])]
     public Collection $entries;
 
     /** @var Collection<int,ActivityEntry> entries */
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply database/models/categories.db.php:31]: I do not think we should change those fields. Also you created a duplicated field with the one below. I suggest we do not change anything here, so you could probably revert the changes to all of the entities.

File: database/models/events.db.php [hunk 3]
```diff
@@ -100,10 +100,6 @@ class Event
     #[OneToMany(targetEntity: EventEntry::class, mappedBy: 'event', cascade: ["remove"])]
     public Collection $entries;
 
-    /** @var Collection<int, Race> */
-    #[OneToMany(targetEntity: Race::class, mappedBy: 'event', cascade: ["remove"])]
-    public Collection $races;
-
```
Comment (Atmos4) [inline database/models/events.db.php:106]: Same here: revert this change (see previous comment)

File: database/models/files.db.php [hunk 4]
```diff
@@ -28,7 +28,7 @@ class SharedFile
     public Permission $permission_level = Permission::USER;
 
     #[ManyToOne]
-    public Race|null $race = null;
```
Comment (Atmos4) [inline database/models/files.db.php:31]: Here we could actually just change that since we are not supposed to use this foreign key yet, but create a migration file for this.

File: database/models/users.db.php [hunk 5]
```diff
@@ -136,7 +132,7 @@ static function getCurrent(): User|null
         if (!has_session("user_id")) {
             return null;
         }
-        if (isset ($_SESSION['controlled_user_id'])) {
```
Comment (Atmos4) [inline database/models/users.db.php:139]: What is this change for?

File: database/models/users.db.php [hunk 6]
```diff
@@ -72,10 +72,6 @@ class User
     #[OneToMany(targetEntity: EventEntry::class, mappedBy: "user", cascade: ["remove"])]
     public Collection $event_entries;
 
-    /** @var Collection<int,RaceEntry> entries */
-    #[OneToMany(targetEntity: RaceEntry::class, mappedBy: "user", cascade: ["remove"])]
-    public Collection $race_entries;
-
```
Comment (Atmos4) [inline database/models/users.db.php:78]: Same here: revert this

File: database/race_to_activity.php [hunk 7]
```diff
@@ -0,0 +1,63 @@
+<?php
+restrict_dev();
+
+$races = em()->createQuery("SELECT r, e.id FROM Race r JOIN Event e")->getArrayResult();
+
+$race_entries = em()->createQuery("SELECT re, c.id FROM RaceEntry re LEFT JOIN re.category c")->getArrayResult();
+
+var_dump($races);
+var_dump($race_entries);
```
Comment (Atmos4) [inline database/race_to_activity.php:9]: Instead of var_dump, use the logger. It prints info in the console in the way that is way more readable
```suggestion
logger()->debug("races", ["races" => $races]);
logger()->debug("raceEntries", ["raceEntries" => $race_entries]);
```

File: database/race_to_activity.php (hunk 1 above)
Comment (Atmos4) [reply database/race_to_activity.php:41]: Let me know if you need some help figuring out what approach to take. Ideally there should only be the need for 1 query here.

File: database/models/files.db.php (hunk 4 above)
Comment (Atmos4) [reply database/models/files.db.php:31]: I would actually suggest dropping this change. We will add a migration in a later PR: This ensures that we dont mess up the DB too much

File: database/models/events.db.php (hunk 3 above)
Comment (Mathieu414) [reply database/models/events.db.php]: ✅

File: database/race_to_activity.php (hunk 1 above)
Comment (Mathieu414) [reply database/race_to_activity.php]: I used the ORM here, thanks for the tips, much easier to work with

File: database/race_to_activity.php (hunk 7 above)
Comment (Mathieu414) [reply database/race_to_activity.php]: Did just that, thx

**Review by Atmos4 — APPROVED** [review]
